redbot/core/utils/common_filters.py

```diff
@@ -16,7 +16,7 @@
 # regexes
 URL_RE = re.compile(r"(https?|s?ftp)://(\S+)", re.I)
 
-INVITE_URL_RE = re.compile(r"(discord\.(?:gg|io|me|li)|discordapp\.com\/invite)\/(\S+)", re.I)
+INVITE_URL_RE = re.compile(r"(discord\.(?:gg|io|me|li)|discord(app)?\.com\/invite)\/(\S+)", re.I)
```

Comment:
The `(app)` group needs to be non-capturing (`(?:app)`), so that developers can use `match.group(1)` and `match.group(2)` to get first and second part of the invite link.

Reply:
Ah, I wasn't familiar with capturing vs non-capturing. Taken care of now